algobattle/match.py

```diff
@@ -195,10 +197,21 @@ def _build(self, teams: list, cache_docker_containers=True) -> bool:
                         process.wait()
                         logger.error('Build process for {} failed!'.format(command[5]))
                         build_successful = False
-            if not build_successful:
-                logger.error("Removing team {} as their containers did not build successfully.".format(team.name))
-                self.team_names.remove(team.name)
-
+                if not build_successful:
+                    logger.error("Removing team {} as their containers did not build successfully.".format(team.name))
+                    self.team_names.remove(team.name)
+                    if command[-2].startswith("generator") and not unsafe_build:
+                        image_archives.pop().unlink()
```

Comment:
If I understand it correctly, we may raise an IndexError here if the very first generator that is being built is not built successfully, as `image_archives` is still empty at this point.

Reply:
image_archives doesn't just contain the paths to generator archives but also solvers, since they are being built first there never will be a generator that is being built first. the intention is to clean up the solver of the team whose generator failed, not to remove another generator.